### PR TITLE
More improvements of GuitarBend user interactions

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #20013 

- Introducing warning colors for bends that are difficult to play (yellow) or that are simply unplayble/incorrect (red).
- Introduging appropriate microtonal accidentals when editing the bend from the property widget onto quarter-values.

### CODE REVIEW

File: src/engraving/dom/guitarbend.cpp
Comment (Eism) [inline src/engraving/dom/guitarbend.cpp:351]: constexpr int UNPLAYABLE_THRESHOLD = 14;
same for warningThreshold

**Review by Eism — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
No way to switch those off like it is possible for notes outside range, the setting that is stored in `MScore::warnPitchRange`?

File: src/engraving/dom/guitarbend.cpp
Comment (Jojo-Schmitz) [inline src/engraving/dom/guitarbend.cpp:584]: No way to switch those off like it is possible for notes outside range, the setting that is stored in MScore::warnPitchRange?
Maybe via this:

`if (score()->printing() || !MScore::warnPitchRange) {`

Comment (mike-spa) [reply src/engraving/dom/guitarbend.cpp:584]: No, we're not bounding this to the same option as the pitch range. It'll probably get its own setting in staff properties instead

**Review by bkunda — COMMENTED** [review]
Nice feature. Works as expected. 

I would suggest we should explore an option for switching off this colouration sooner rather than later. It should probably live in Preferences (beneath `Colour notes outside of usable pitch range`) because the user goal is generally the same for both: I.e. "When entering notation, I don't want the program to colour notation that it thinks isn't playable – I just want to work with a black and white score".  I'd suggest labelling it `Colour guitar bends outside of playable range`.

**Comment by bkunda** [comment]
I'm thinking the checkbox should probably be checked by default. Otherwise the coloration feature will be completely undiscoverable.

**Comment by Jojo-Schmitz** [comment]
FWIW: I agree and was tempted to make the same comment

**Comment by mike-spa** [comment]
> I'm thinking the checkbox should probably be checked by default. Otherwise the coloration feature will be completely undiscoverable.

@bkunda that's what I actually meant to do, my mistake

**Comment by mike-spa** [comment]
Btw, the vtests failures are due to other changes unrelated to this PR. So this is good for merging from my side, if the functionality is ok :)

**Comment by RomanPudashkin** [comment]
@mike-spa @Eism @bkunda let's not merge this PR until [my PR is merged](https://github.com/musescore/MuseScore/pull/19921) to avoid more conflicts (or at least move the changes in bendsettingsmodel.cpp to a separate PR)

File: src/engraving/dom/guitarbend.cpp
Comment (RomanPudashkin) [inline src/engraving/dom/guitarbend.cpp:582]: uiColor

**Comment by mike-spa** [comment]
@RomanPudashkin @Eism please take a good look at the code now. There were a lot of conflicts, so I have done my best to make sure that the correct version of each piece of code was kept, let me know if I've done something wrong. The PR is now rebased and, as suggested by @bkunda, I have rolled back coloring the widget curve (meaning that now only the actual notated bend wil turn yellow/red when needed, not the widget curve).

File: src/inspector/models/notation/bends/bendsettingsmodel.cpp
Comment (Eism) [inline src/inspector/models/notation/bends/bendsettingsmodel.cpp:273]: please use endTimePoint here

**Comment by Eism** [comment]
@mike-spa 

<details>
<summary>in BendGridCanvas::movePoint should be this code</summary>

```
if (canMoveVertically) {
        bool canMove = true;

        if (currentPoint.limitMoveVerticallyByNearestPoints) {
            bool moveToTop = currentPoint.pitch < toPoint.pitch;
            if (pointIndex - 1 >= 0) {
                const CurvePoint& leftPoint = m_points.at(pointIndex - 1);
                bool isLeftValid = moveToTop ? leftPoint.pitch >= currentPoint.pitch : leftPoint.pitch <= currentPoint.pitch;
                if (isLeftValid) {
                    canMove = leftPoint.generated || (moveToTop ? leftPoint.pitch > toPoint.pitch : leftPoint.pitch < toPoint.pitch);
                }
            }

            if (!canMove) {
                return moved;
            }

            if (pointIndex + 1 < m_points.size()) {
                const CurvePoint& rightPoint = m_points.at(pointIndex + 1);
                bool isRightValid = moveToTop ? rightPoint.pitch >= currentPoint.pitch : rightPoint.pitch <= currentPoint.pitch;
                if (isRightValid) {
                    canMove = rightPoint.generated || (moveToTop ? rightPoint.pitch > toPoint.pitch : rightPoint.pitch < toPoint.pitch);
                }
            }
        }

        if (canMove) {
            currentPoint.pitch = toPoint.pitch;

            bool isDashed = currentPoint.endDashed;
            bool isNextDashed = (pointIndex + 1 < m_points.size()) && m_points.at(pointIndex + 1).endDashed;

            if (isDashed) {
                m_points[pointIndex - 1].pitch = toPoint.pitch;
            }

            if (isNextDashed) {
                m_points[pointIndex + 1].pitch = toPoint.pitch;
            }

            moved = true;
        }
    }

    if (canMoveHorizontally) {
        bool canMove = true;

        bool moveToLeft = currentPoint.time > toPoint.time;
        if (moveToLeft) {
            if (pointIndex - 1 >= 0) {
                const CurvePoint& leftPoint = m_points.at(pointIndex - 1);
                canMove = leftPoint.generated || leftPoint.time < toPoint.time;
            }
        } else {
            if (pointIndex + 1 < m_points.size()) {
                const CurvePoint& rightPoint = m_points.at(pointIndex + 1);
                canMove = rightPoint.generated || rightPoint.time > toPoint.time;
            }
        }

        if (canMove) {
            currentPoint.time = toPoint.time;
            moved = true;
        }
    }
```

</details>

File: src/engraving/dom/guitarbend.cpp
Comment (RomanPudashkin) [inline src/engraving/dom/guitarbend.cpp:96]: quarterOff should have a different type (enum or bool)

**Review by Eism — APPROVED** [review]

**Review by bkunda — APPROVED** [review]

**Pull request merged**
